src/EditorFeatures/CSharp/MisplacedUsings/MisplacedUsingsDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,141 @@
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class MisplacedUsingsDiagnosticAnalyzer : AbstractCodeStyleDiagnosticAnalyzer
+    {
+        private static readonly CodeStyleOption<UsingPlacementPreference> s_noPreferenceOption =
+            new CodeStyleOption<UsingPlacementPreference>(UsingPlacementPreference.NoPreference, NotificationOption.None);
+
+        private static readonly LocalizableResourceString s_localizableTitle = new LocalizableResourceString(
+            nameof(CSharpEditorResources.Misplaced_using), CSharpEditorResources.ResourceManager, typeof(CSharpEditorResources));
+
+        private static readonly LocalizableResourceString s_localizableInsideMessage = new LocalizableResourceString(
+            nameof(CSharpEditorResources.Using_directives_must_be_placed_inside_a_namespace_declaration), CSharpEditorResources.ResourceManager, typeof(CSharpEditorResources));
+
+        private static readonly LocalizableResourceString s_localizableOutsideMessage = new LocalizableResourceString(
+            nameof(CSharpEditorResources.Using_directives_must_be_placed_outside_a_namespace_declaration), CSharpEditorResources.ResourceManager, typeof(CSharpEditorResources));
+
+        internal static readonly DiagnosticDescriptor _insideDescriptor = CreateDescriptorWithId(
+            IDEDiagnosticIds.MoveMisplacedUsingsDiagnosticId,
+            s_localizableTitle,
+            s_localizableInsideMessage,
+            configurable: true);
+
+        internal static readonly DiagnosticDescriptor _outsideDescriptor = CreateDescriptorWithId(
+            IDEDiagnosticIds.MoveMisplacedUsingsDiagnosticId,
+            s_localizableTitle,
+            s_localizableOutsideMessage,
+            configurable: true);
+
+        private readonly ICodingConventionsManager _conventionsManager = CodingConventionsManagerFactory.CreateCodingConventionsManager();
+
+        public MisplacedUsingsDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.MoveMisplacedUsingsDiagnosticId, s_localizableTitle)
+        {
+        }
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.CompilationUnit);
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.NamespaceDeclaration);
+        }
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var statement = context.Node;
+            var cancellationToken = context.CancellationToken;
+
+            var option = GetPreferredPlacementOption(context);
+            if (option.Value == UsingPlacementPreference.NoPreference)
+            {
+                return;
+            }
+
+            if (statement is CompilationUnitSyntax compilationUnit
+                && option.Value == UsingPlacementPreference.InsideNamespace
+                && !ShouldSurpressDiagnostic(compilationUnit))
```

Comment:
Was trying to remove duplication of the code gets the placement preference since it is a majority of the code, but i see now it wasn't an issue. =) Thanks!